kolibri/plugins/slideshow_render/assets/src/views/SlideshowRendererComponent.vue

```diff
@@ -0,0 +1,214 @@
+
+  import UiIconButton from 'kolibri.coreVue.components.UiIconButton';
+  import CoreFullscreen from 'kolibri.coreVue.components.CoreFullscreen';
+  import {
+    Hooper,
+    Slide,
+    Navigation as HooperNavigation,
+    Pagination as HooperPagination,
+  } from 'hooper';
+  import '../../styles/custom-hooper.css';
+
+  export default {
+    name: 'SlideshowRendererComponent',
+    components: {
+      UiIconButton,
+      CoreFullscreen,
+      Hooper,
+      Slide,
+      HooperPagination,
+      HooperNavigation,
+    },
+    mixins: [contentRendererMixin, themeMixin, responsiveElement, responsiveWindow],
+    props: {
+      defaultFile: {
+        type: Object,
+        default: () => {},
+      },
+      files: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    data: () => ({
+      isInFullscreen: false,
+      manifest: {},
+      slides: null,
+      currentSlide: null,
+      hooperSettings: {
+        centerMode: true,
+      },
+    }),
+    computed: {
+      slideshow_images: function() {
+        const files = this.files;
+        return files.filter(file => file.preset != 'Slideshow Manifest');
+      },
+      caption_height: function() {
+        return 30 + (this.currentSlide ? this.$refs[this.currentSlide.id][0].clientHeight : 0);
+      },
+    },
+    mounted() {
+      /*
+        Using the manifest file, get the JSON from the manifest, then
+        use the manifest JSON to get all slide images and metadata.
+      */
+      fetch(this.defaultFile.storage_url, { method: 'GET' })
+        .then(response => {
+          return response.status === 200 ? response.json() : {};
+        })
+        .then(manifest_data => {
+          this.manifest = manifest_data.slideshow_data;
+
+          this.slides = _.orderBy(
+            this.manifest.map(image => {
+              return {
+                storage_url: this.slideshow_images.find(
+                  sFile =>
+                    sFile.storage_url
+                      .split('/')
+                      [sFile.storage_url.split('/').length - 1].split('.')[0] == image.checksum
+                ).storage_url,
+                caption: image.caption,
+                sort_order: image.sort_order,
+                id: image.checksum,
+                descriptive_text: image.descriptive_text,
+              };
+            }),
+            ['sort_order'],
+            ['asc']
+          );
+        });
+    },
+    methods: {
+      handleSlide(payload) {
+        this.currentSlide = this.slides[payload.currentSlide];
+      },
+      slideTextId(id) {
+        return 'descriptive-text-' + id;
+      },
+    },
+    $trs: {
+      exitFullscreen: 'Exit fullscreen',
+      enterFullscreen: 'Enter fullscreen',
+    },
+  };
+
+</script>
+
+
+<style lang="scss" scoped>
+
+  .btn {
+    position: absolute;
+    top: 8px;
+    right: 8px;
+    z-index: 100;
+    fill: white;
+  }
```

Comment:
suuuper minor but most other places in the code we put a line break between class definitions